File: flutter_application_signalr_chat_test/lib/Entities/DeviceTypes.dart
class DeviceTypes {
  final int deviceTypeID;
  final String deviceTypeName;
  final String imagePath;

  DeviceTypes({
    required this.deviceTypeID,
    required this.deviceTypeName,
    required this.imagePath,
  });

  factory DeviceTypes.fromJson(Map<String, dynamic> json) {
    return DeviceTypes(
      deviceTypeID: json['DeviceTypeID'] is int ? json['DeviceTypeID'] : (json['DeviceTypeID'] as num?)?.toInt() ?? 0,
      deviceTypeName: json['DeviceTypeName'] ?? 'Unknown',
      imagePath: json['ImagePath'] ?? '',
    );
  }


  Map<String, dynamic> toJson() {
    return {
      'DeviceTypeID': deviceTypeID,
      'DeviceTypeName': deviceTypeName,
      'ImagePath': imagePath,
    };
  }
}

File: Server/Server_Test_SignalR/Server_Test_SignalR/Entieis/DeviceTypes.cs
using Newtonsoft.Json;
using System.Security.Cryptography;

namespace Server_Test_SignalR.Entieis
{
    public class DeviceTypes
    {
        public long DeviceTypeID { get; set; }
        public string DeviceTypeName { get; set; }
        public string ImagePath { get; set; }

        public virtual void PrintDeviceInfo()
        {
            Console.WriteLine($"DeviceType ID: {DeviceTypeID}");
            Console.WriteLine($"DeviceType Name: {DeviceTypeName}");
            Console.WriteLine($"Image Path: {ImagePath}");
        }

        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }
    }
}
